daemon/controller.cpp

```diff
@@ -108,6 +108,17 @@ Sailfish::Secrets::Daemon::Controller::crypto() const
     return m_crypto;
 }
 
+QWeakPointer<QThreadPool> Sailfish::Secrets::Daemon::Controller::threadPoolForPlugin(const QString &pluginName) const
+{
+    if (m_secrets->potentialCryptoStoragePlugins().contains(pluginName)) {
```

Comment:
Does the order matter here?

Reply:
Yes.  If a crypto plugin is also an encrypted storage plugin, then all its operations must be run on the Secrets Plugin thread.